src/test/ui/or-patterns/or-patterns-syntactic-fail.rs

```diff
@@ -8,7 +8,9 @@ use E::*;
 
 fn no_top_level_or_patterns() {
     // We do *not* allow or-patterns at the top level of lambdas...
-    let _ = |A | B: E| (); //~ ERROR no implementation for `E | ()`
+    let _ = |A | B: E| ();
+    //~^ ERROR mismatched types
+    //~^^ ERROR no implementation for `E | NonZero<E>`
```

Comment:
Same goes for this, but this seems to be cause by the same problem related to `BitOr`.